CMake/RealmCore.cmake

```diff
@@ -238,25 +237,41 @@ endfunction()
 
 macro(build_realm_core)
     set(core_prefix_directory "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/realm-core")
-    ExternalProject_Add(realm-core
-        PREFIX ${core_prefix_directory}
-        BUILD_IN_SOURCE 1
-        BUILD_COMMAND make -C src/realm librealm.a librealm-dbg.a ${MAKE_FLAGS}
-        INSTALL_COMMAND ""
-        CONFIGURE_COMMAND ""
-        ${USES_TERMINAL_BUILD}
-        ${ARGV}
-        )
+    if(REALM_PLATFORM STREQUAL "Android")
+      ExternalProject_Add(realm-core
+            PREFIX ${core_prefix_directory}
+            BUILD_IN_SOURCE 1
+            BUILD_COMMAND sh build.sh build-android
+            INSTALL_COMMAND ""
+            CONFIGURE_COMMAND REALM_ENABLE_ENCRYPTION=YES REALM_ENABLE_ASSERTIONS= sh build.sh config
+            ${USES_TERMINAL_BUILD}
+            ${ARGV}
+            )
+    else()
+        ExternalProject_Add(realm-core
+            PREFIX ${core_prefix_directory}
+            BUILD_IN_SOURCE 1
+            BUILD_COMMAND make -C src/realm librealm.a librealm-dbg.a ${MAKE_FLAGS}
+            INSTALL_COMMAND ""
+            CONFIGURE_COMMAND ""
+            ${USES_TERMINAL_BUILD}
+            ${ARGV}
+            )
+    endif()
 
     ExternalProject_Get_Property(realm-core SOURCE_DIR)
     define_realm_core_target(NO ${SOURCE_DIR})
 endmacro()
 
 function(clone_and_build_realm_core branch)
+    if(REALM_PLATFORM STREQUAL "Android")
+        set(CONFIG_CMD REALM_ENABLE_ENCRYPTION=YES REALM_ENABLE_ASSERTIONS= sh build.sh config)
+    else()
+        set(CONFIG_CMD REALM_ENABLE_ENCRYPTION=YES REALM_ENABLE_ASSERTIONS=YES sh build.sh config)
+    endif()
     build_realm_core(GIT_REPOSITORY "https://github.com/realm/realm-core.git"
                      GIT_TAG ${branch}
-                     CONFIGURE_COMMAND test -f src/config.mk ||
-                                       REALM_ENABLE_ENCRYPTION=YES REALM_ENABLE_ASSERTIONS=YES sh build.sh config
+                     CONFIGURE_COMMAND test -f src/config.mk || CONFIG_CMD
```

Comment:
This is currently breaking the build for this branch since you're not expanding the variable.